src/coreclr/vm/interpframeallocator.cpp

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifdef FEATURE_INTERPRETER
+
+#include "interpexec.h"
+#include "interpframeallocator.h"
+
+FrameDataFragment::FrameDataFragment(size_t size)
+{
+    if (size < INTERP_STACK_FRAGMENT_SIZE)
+    {
+        size = INTERP_STACK_FRAGMENT_SIZE;
+    }
+
+    start = (uint8_t*)malloc(size);
+    if (start == nullptr)
+    {
+        // Interpreter-TODO: Throw OutOfMemory exception
```

Comment:
I expect that the managed exception is going to be actually thrown interpexec.cpp as there won't be an infrastructure to throw the managed exceptions from outside of interpexec.cpp.

I think it would be better to return null from here (it won't be able to be a constructor anymore), and add this TODO to interpexec.cpp where the exception will be actually thrown eventually.

Reply:
Sounds good, updated.